core/src/main/kotlin/org/kkafka/consumer/commit.kt

```diff
@@ -0,0 +1,43 @@
+package org.kkafka
+
+import kotlin.Result.Companion.failure
+import kotlin.Result.Companion.success
+import kotlin.coroutines.suspendCoroutine
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.clients.consumer.OffsetAndMetadata
+import org.apache.kafka.common.TopicPartition
+
+/**
+ * Like [KafkaConsumer.commitAsync], except
+ * - It suspends rather than take a callback
+ * - It throws rather than pass a nullable exception in the callback
+ * - It returns the offsets passed to the callback
+ */
+public suspend fun <K, V> KafkaConsumer<K, V>.commit(): Map<TopicPartition, OffsetAndMetadata> =
```

Comment:
I agree with Arman it might be better to call it commitSuspend so we have the explicit three ways of committing with sync, async and suspend

Reply:
I only half agree, because `suspend` is part of the type signature already

addressed